apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/IconCatalog/IconCatalog.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { Provider, teamsV2Theme } from '@fluentui/react-northstar';
+import * as v0Icons from '@fluentui/react-icons-northstar';
+import * as v9Icons from '@fluentui/react-icons';
+import { makeStyles, Input, Switch, Label } from '@fluentui/react-components';
+import type { InputProps, SwitchProps } from '@fluentui/react-components';
+import { iconMapping as rawMapping } from './iconMapping';
+import { IconGrid } from './IconGrid';
+import { useDebounce } from './useDebounce';
+import { V0IconComponent, V9IconComponent } from './types';
+
+const useStyles = makeStyles({
+  searchPanel: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    marginBottom: '10px',
+    marginTop: '20px',
+  },
+
+  switch: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+});
+
+const mapping = rawMapping
+  .map(entry => {
+    const v0IconName = `${entry.v0}`;
+    const v9IconName = `${entry.v9}Regular`;
+    // @ts-expect-error - We want to index the exports by name here
+    const V0Icon = v0Icons[v0IconName] as V0IconComponent;
+    // @ts-expect-error - We want to index the exports by name here
+    const V9Icon = v9Icons[v9IconName] as V9IconComponent;
+
+    if (!V0Icon) {
+      return null;
+    }
+
+    return {
+      v9Search: entry.v9?.toLowerCase(),
+      v0Search: entry.v0.toLowerCase(),
+      V0Icon,
+      V9Icon,
+    };
+  })
+  .filter(Boolean) as { v9Search: string; v0Search: string; V0Icon: V0IconComponent; V9Icon: V9IconComponent }[];
+
+const IconCatalogInner: React.FC = () => {
+  const styles = useStyles();
+  const [searchTerm, setSearchTerm] = React.useState<string | undefined>(undefined);
+  const [searchV0, setSearchV0] = React.useState<boolean>(true);
+
+  const updateSearch = React.useCallback(
+    (newSearchTerm: string) => {
+      setSearchTerm(newSearchTerm.toLowerCase());
+    },
+    [setSearchTerm],
+  );
+
+  const updateSearchDebounced = useDebounce(updateSearch, 220);
+
+  const entries = React.useMemo(
+    () =>
+      mapping.filter(entry => {
+        if (!searchTerm) {
+          return true;
+        }
+
+        if (searchV0) {
+          return entry.v0Search.includes(searchTerm);
+        }
+
+        return entry.v9Search?.includes(searchTerm);
+      }),
+    [searchTerm, searchV0],
+  );
+
+  const onInputChange: InputProps['onChange'] = React.useCallback((e, { value }) => updateSearchDebounced(value), [
+    updateSearchDebounced,
+  ]);
+  const onSwitchChange: SwitchProps['onChange'] = React.useCallback((e, { checked }) => setSearchV0(checked), []);
+
+  return (
+    <>
+      <div>
+        <div className={styles.searchPanel}>
```

Comment:
Would great to include a section on what to do with props from N* icons and explain how to bundle icons (link to icons page?)

![image](https://user-images.githubusercontent.com/14183168/196212313-ed68b5e3-22d1-41de-a744-f0109248fb28.png)